src/scripts/ui/system-status.js

```diff
@@ -0,0 +1,98 @@
+window.addEventListener('DOMContentLoaded', () => {
+  const $headerStatus = document.querySelector('.header-system-status-container');
+  const $headerStatusMessage = document.querySelector('.header__status-message');
+
+  const $operationalIcons = document.querySelectorAll('.operational');
+  const $degradedPerformanceIcons = document.querySelectorAll('.degraded-performance');
+  const $partialServiceDisruptionIcons = document.querySelectorAll('.partial-service-disruption');
+  const $serviceDisruptionIcons = document.querySelectorAll('.service-disruption');
+  const $securityEventIcons = document.querySelectorAll('.security-event');
+  const $allSystemStatusIcons = document.querySelectorAll('.system-status');
+
+  // https://kb.status.io/developers/status-codes/
+  const status = {
+    DEGRADED_PERFORMANCE: {
+      icons: $degradedPerformanceIcons,
+      message: 'Degraded Performance - We are currently experiencing degraded performance in some of our web services.',
+    },
+    OPERATIONAL: {
+      icons: $operationalIcons,
+      message: '',
+    },
+    PARTIAL_SERVICE_DISRUPTION: {
+      icons: $partialServiceDisruptionIcons,
+      message: 'Partial Service Disruption - Some of our web services are temporarily unavailable.',
+    },
+    SECURITY_EVENT: {
+      icons: $securityEventIcons,
+      message: 'Security Event - We are currently mitigating issues relating to some of our web services.',
+    },
+    SERVICE_DISRUPTION: {
+      icons: $serviceDisruptionIcons,
+      message: 'Service Disruption - Our web services are temporarily unavailable.',
+    },
+  };
+
+  const setSystemStatus = ($icons, message) => {
+    $allSystemStatusIcons.forEach($systemStatusIcon => {
+      if ($systemStatusIcon) {
+        $systemStatusIcon.classList.remove('show-status-icon');
+        $systemStatusIcon.classList.add('status-icon');
```

Comment:
Adding and removing `status-icon` shouldn't be necessary, as `show-status-icon` has `display: block !important` which overrides `status-icon`'s `display:none`.

You can then use `status-icon` instead of `system-status` for `$allSystemStatusIcons`